oozie/src/main/java/org/apache/falcon/oozie/ImportExportCommon.java

```diff
@@ -129,4 +129,45 @@ public static StringBuilder buildTableArg(StringBuilder builder, String tableNam
         return builder.append("--table").append(ImportExportCommon.ARG_SEPARATOR)
                 .append(tableName);
     }
+
+    public static StringBuilder buildArguments(StringBuilder builder, Map<String, String> extraArgs)
+            throws FalconException {
+        handleVerbose(builder, extraArgs);
+        for(Map.Entry<String, String> e : extraArgs.entrySet()) {
+            if (!e.getKey().startsWith("--direct")) {
+                builder.append(e.getKey()).append(ImportExportCommon.ARG_SEPARATOR).append(e.getValue())
+                        .append(ImportExportCommon.ARG_SEPARATOR);
+            }
+        }
+        return builder;
+    }
+
+    public static StringBuilder handleVerbose(StringBuilder builder, Map<String, String> extraArgs)
+            throws FalconException {
+        if ((extraArgs.containsKey("--verbose")) && (extraArgs.get("--verbose").equalsIgnoreCase("TRUE"))) {
+            builder.append("--verbose").append(ImportExportCommon.ARG_SEPARATOR);
+            extraArgs.remove("--verbose");
+        }
+        return builder;
+    }
+
+    public static StringBuilder handleDirectMode(StringBuilder builder, Map<String, String> extraArgs)
+            throws FalconException {
+        if ((extraArgs.containsKey("--direct")) && (extraArgs.get("--direct").equalsIgnoreCase("true"))) {
+            builder.append("--direct").append(ImportExportCommon.ARG_SEPARATOR);
+        }
+        boolean first = true;
+        for(Map.Entry<String, String> e : extraArgs.entrySet()) {
+            if (e.getKey().startsWith("--direct.")) {
+                if (first) {
+                    builder.append("--").append(ImportExportCommon.ARG_SEPARATOR);
```

Comment:
all other options are added that way (see below) in the XML. 
                    <argument name="--split-by" value="id"/>
                    <argument name="--num-mappers" value="2"/>
                    <argument name="--direct" value="true"/>
                    <argument name="--direct.default-character-set" value="latin1"/>
                    <argument name="--verbose" value="true"/>